### ISSUE TITLE
mat-tree: table layout

### ISSUE THREAD

**Issue by volago** [angular/material2#13616]
#### feature request:
It would be great to style mat tree like a table. Mat-tree is very convenient to deal with hierarchical structures, but sometimes one want to display hierarchy as a table, See example from DexExpress guys: https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeList/Overview/Angular/Light/

**Comment by djleonskennedy** [comment]
+1

**Comment by jelbourn** [comment]
For this type of interaction, we think the right approach is to use the table component with some whitespace in the first column.

**Comment by volago** [comment]
Hi. It's hard to follow your advice. I think it is far easier to style mat-tree rather than add treeControl functionality to mat-table. Finally I achieved what I wanted by following code using flex-layout lib:

```html
<div>
    <table class="mat-table">
        <thead class="mat-header-row" fxLayoutAlign=" center">
            <th class="mat-header-cell" fxFlex>
                ...
            </th>
            <th class="mat-header-cell" fxFlex="20">
                ...
            </th>
            <th class="mat-header-cell" fxFlex="20">
                ...
            </th>
            <th class="mat-header-cell" fxFlex="20">
                ...
            </th>
            <th class="mat-header-cell" fxFlex="10"></th>
        </thead>
    </table>

    <mat-tree [dataSource]="dataSource" [treeControl]="treeControl">
        <mat-tree-node *matTreeNodeDef="let node" class="border-top">
            <div fxFlex fxLayoutAlign=" center">
                  <span *ngFor="let i of array(treeControl.getLevel(node))" class="indent"></span>
                ...
            </div>
            <div fxFlex="20" class="text-center">
                ...
            </div>
            <div fxFlex="20" class="text-center">
                ...
            </div>
            <div fxFlex="20" class="text-center">...</div>
            <div fxFlex="10" class="text-center">
                ...
            </div>
        </mat-tree-node>
        		[ mat-tree-node for children ]	
    </mat-tree>
	
</div>
```

**Comment by yd021976** [comment]
@volago Hello, I need the same layout (i.e Table heads for tree view).
As I'm not HTML/CSS expert, do you have a sample working code that show how you use Flex-layout lib and Material Tree ?

Thank you

**Comment by volago** [comment]
@yd021976 Hi. I have this working, but in the production code, so I cannot share it with you ;) If you want you can create a stackblitz sample (starting i.e. form https://material.angular.io/components/tree/examples) and then I can help you to make it work if you'll have any problems.

**Comment by yd021976** [comment]
@volago , Hi and thank you for your help. Here is a quick stackblitz :[Example with flexlayout](https://angular-zp4xqe.stackblitz.io)

I'm simply starting this example by copying Material tree example "tree with nested node" and try to add flex layout to tree nodes (never mind the table column header names, it's just for example)

the main problem is when I add "flex layout", the tree do not expand anymore (or I can't see the expanded content)

Later I'll try to align table header with tree nodes (I'll have to learn flex layout !)

Thanks again

**Comment by volago** [comment]
It turned out that not every example I recommended is suitable for our table-tree-control. Apologies.

I picked up the proper one and made it working here: https://stackblitz.com/angular/bamjqopeyxd.
You need to remember to add @angular/flex-layout package and (!) import it in a module.

Hope it helps

**Comment by sam1721** [comment]
Is it possible only to use mat-table and achieve this without mat-tree and flex layout package. In my case, I can have arbitrarily number of hierarchical depth that the row can be grouped and there are items which aren't grouped as well, like there can be 2 items that can be grouped and rest 2 which aren't grouped and item that aren't grouped can be reorder. So I needed to achieve this only by using mat-table if its possible.

**Comment by volago** [comment]
@sam1721 great to hear this, could you show an example (stackblitz prefered)
